.github/workflows/main.yml

```diff
@@ -87,8 +86,8 @@ jobs:
         id: docker_build
         uses: docker/build-push-action@v2
         with:
-          context: .
-          push: true
+          platforms: linux/amd64,linux/arm64/v7,linux/arm64/v8
```

Comment:
I was searching for the available list of platforms and I noticed there is a difference with and without the qemu step. This is with qemu:

![image](https://user-images.githubusercontent.com/165497/129597127-cea29753-1a3e-4eb8-8ea4-d0d0343185cf.png)

> Platforms: linux/amd64, linux/arm64, linux/riscv64, linux/ppc64le, linux/s390x, linux/386, linux/mips64le, linux/mips64, linux/arm/v7, linux/arm/v6

With qemu removed (in this pull request), it changed to:

![image](https://user-images.githubusercontent.com/165497/129597300-9be20a71-1bf9-46de-8684-5bcc1dba62a4.png)

> Platforms: linux/amd64, linux/386

Should the qemu step be added back again? And would it make sense to update the platforms to:

- linux/amd64 (desktop and servers / what is being published now to Docker Hub)
- linux/arm/v6 (Older Raspberry Pi models)
- linux/arm/v7 (Recent Raspberry Pi models)
- linux/arm64 (ARM 64 bit)

Reply:
Qemu is needed only when want emulation but in our case with `--platform=$BUILDPLATFORM` in the docker file emulation is not needed as this flags selects the image for the required platform. 

don't think should support v6 as the performance there would probably be suboptimal.

Reply:
While v6 might not be optimal, I think it is good to support it, as it would be needed for the Raspberry Pi Zero W for example. At some point I would consider integrating containers within the ChirpStack Gateway OS, so that it becomes easier to upgrade individual components rather than upgrading the complete OS each time + it would add the possibility to side-load additional software.

> Qemu is needed only when emulation is needed

Are you sure? Looking at your example build (https://github.com/arribada/chirpstack-application-server/runs/3320609454?check_suite_focus=true#step:6:103), it looks like `linux/arm64/v7` was dropped, see the arguments on line 103: `--platform linux/amd64,linux/arm64/v8`. My assumption is that this is because v7 is 32 bit and should thus have been `linux/arm/v7`?

But also, looking further in the logs, I only see references to `linux/amd64` being built, e.g. https://github.com/arribada/chirpstack-application-server/runs/3320609454?check_suite_focus=true#step:6:386. I can not find any further references to `linux/arm64/v8`.

I would expect that for the ARM builds, the `docker build ...` would need to run in a emulated environment, as the `FROM --platform=$BUILDPLATFORM golang:1.16-alpine` would fetch the ARM version of the `golang:1.16-alpine` image, and thus the build steps would fail on an amd64 environment as the `go` binary within that image can only be executed on an ARM architecture?

Please let me know if I'm missing something.

Reply:
hhm, I think you are right I re-added qemu and lets see the logs from latest run in this PR

Reply:
The platforms `linux/arm64/v6` and `linux/arm64/v7` should be `linux/arm/v6` and `linix/arm/v7`, respectively

Reply:
thanks you are right I updated the references. Can see the avail platforms in the logs.


> But also, looking further in the logs, I only see references to linux/amd64 being built, e.g. arribada/chirpstack-application-server/runs/3320609454?check_suite_focus=true#step:6:386. I can not find any further references to linux/arm64/v8.

not sure about this one with or without qemu it only shows logs regarding the amd64 build. Even the raw logs don't show anything different

Reply:
I think I figured it out, the extra things I added to the docker file are the cause.

Reply:
Yes all good, the last build shows the logs for all architectures.